components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/ApplicationAuthenticatorService.java

```diff
@@ -88,6 +91,10 @@ public RequestPathAuthenticatorConfig getRequestPathAuthenticatorByName(String n
 
     public void addLocalAuthenticator(LocalAuthenticatorConfig authenticator) {
         if (authenticator != null) {
+            //TODO: Remove warn log, once feature is ready.
+            if (authenticator.getDefinedByType() == null) {
+                LOG.warn("The defined by type is not set for the : " + authenticator.getName());
+            }
```

Comment:
Well, IMO this is not a good practice to follow.
We clearly know that this can be null for exisitng authenticators, so we should move to the default which is system

Reply:
Addressed